src/msspreadsheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,639 @@
+        Workbook workBook = new(self.microsoftGraphConfig, path, workbookName);
+
+        return workBook;
+    }
+};
+
+# Workbook Client Object.
+# + workbookClient - HTTP client endpoint for the workbook
+# + properties - Workbook specific properties
+# + microsoftGraphConfig - Configurations for accessing spreadsheet API
+public type Workbook client object {
+    http:Client workbookClient;
+    WorkbookProperties properties = {"path":"", "workbookName":""};
+    MicrosoftGraphConfiguration microsoftGraphConfig;
+
+    public function __init(MicrosoftGraphConfiguration msGraphConfig, string path, string workbookName) {
+        self.microsoftGraphConfig = msGraphConfig;
+        self.properties = {"path":path, "workbookName":workbookName};
+
+        oauth2:OutboundOAuth2Provider oauth2Provider3 = new ({
+            accessToken: msGraphConfig.msInitialAccessToken,
+            refreshConfig: {
+                clientId: msGraphConfig.msClientID,
+                clientSecret: msGraphConfig.msClientSecret,
+                refreshToken: msGraphConfig.msRefreshToken,
+                refreshUrl: msGraphConfig.msRefreshURL,
+                clientConfig: {
+                    secureSocket: {
+                        trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                        }
+                    }
+                }
+            }
+        });
+
+        http:BearerAuthHandler oauth2Handler3 = new (oauth2Provider3);
+
+        self.workbookClient = new (msGraphConfig.baseUrl, {
+            auth: {
+                authHandler: oauth2Handler3
+            },
+            secureSocket: {
+                trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                }
+            }
+        });
+    }
+
+    # Get the properties of the workbook.
+    # + return - Properties of the Workbook
+    public function getProperties() returns WorkbookProperties {
+        return self.properties;
+    }
+
+    # Open a worksheet on this workbook.
+    # + worksheetName - name of the worksheet to be opened
+    # + return - A Worksheet client object on success, else returns an error
+    public remote function openWorksheet(string worksheetName) returns @tainted (Worksheet|error) {
+        http:Request request = new;
+        http:Response|error response = self.workbookClient->get("/v1.0/me/drive/root:" + self.properties.path +
+                            self.properties.workbookName + ".xlsx:/workbook/worksheets/" + worksheetName, request);
+
+        if response is error {
+            HttpError httpError = error(HTTP_ERROR, message = "Error occurred while accessing the Microsoft Graph API.", 
+                                        errorCode = HTTP_ERROR, cause = response);
```

Comment:
Is there a reason to add `HTTP_ERROR` twice in the error record? We already add it as the reason.
If there's no usage, shall we remove it? Check the other places as well.

Reply:
I fixed this issue in all the places it appeared